cetlvast/suites/unittest/test_variable_length_array_bool.cpp

```diff
@@ -231,4 +231,112 @@ TYPED_TEST(VLABoolTests, TestBoolIterator)
     ASSERT_EQ(false, *foo.cbegin());
     ASSERT_EQ(true, *(foo.cend() - 1));
     ASSERT_EQ(true, foo.cbegin()[2]);
+
+    typename decltype(foo)::const_iterator const_begin_iterator =
+        const_cast<typename std::add_pointer<typename std::add_const<decltype(foo)>::type>::type>(&foo)->begin();
+    typename decltype(foo)::const_iterator const_end_interator =
+        const_cast<typename std::add_pointer<typename std::add_const<decltype(foo)>::type>::type>(&foo)->end();
+
+    ASSERT_EQ(false, *const_begin_iterator);
+    ASSERT_EQ(true, *(const_end_interator - 1));
+    ASSERT_EQ(true, const_begin_iterator[2]);
+}
+
+TYPED_TEST(VLABoolTests, TestBoolPopBack)
+{
+    auto test_subject = TypeParam::make_bool_container(
+        std::initializer_list<bool>{true, false, true, false, true, false, true, false, true});
+    std::size_t starting_size = test_subject.size();
+    ASSERT_EQ(9, starting_size);
+    while (starting_size > 0)
+    {
+        ASSERT_EQ((starting_size % 2), test_subject[starting_size - 1]);
+        test_subject.pop_back();
+        --starting_size;
+        ASSERT_EQ(starting_size, test_subject.size());
+    }
+}
+
+TYPED_TEST(VLABoolTests, TestBoolResize)
+{
+    auto array = TypeParam::make_bool_container();
+    ASSERT_EQ(0, array.size());
+    for (std::size_t i = 1; i <= 64; ++i)
+    {
+        array.resize(i, false);
+        ASSERT_EQ(i, array.size());
+        ASSERT_EQ(false, array[i - 1]);
+    }
+}
+
+TYPED_TEST(VLABoolTests, TestBoolResizeToZero)
+{
+    auto test_subject = TypeParam::make_bool_container(
+        std::initializer_list<bool>{true, false, true, false, true, false, true, false, true});
+    ASSERT_EQ(9, test_subject.size());
+    test_subject.resize(0);
+    ASSERT_EQ(0, test_subject.size());
+    ASSERT_LE(9, test_subject.capacity());
+}
+
+TYPED_TEST(VLABoolTests, TestBoolResizeWithDefault)
+{
+    auto array = TypeParam::make_bool_container(std::initializer_list<bool>{false});
+    array.resize(22, true);
+    ASSERT_EQ(22, array.size());
+    ASSERT_EQ(false, array[0]);
+    for (std::size_t i = 1; i < array.size(); ++i)
+    {
+        ASSERT_EQ(true, array[i]);
+    }
+    // Remember that, if we resize down, the default argument is not used for anything.
+    array.resize(9, false);
+    ASSERT_EQ(9, array.size());
+    ASSERT_EQ(false, array[0]);
+    for (std::size_t i = 2; i < array.size(); ++i)
+    {
+        ASSERT_EQ(true, array[i - 1]);
```

Comment:
Seems like loop and index should match earlier loop and index (i.e. start at i=1 and array[i])

Reply:
Good call. It's correct but wonky. Will fix.